From 586638064830c64e353a00176182e056c0b459e7 Mon Sep 17 00:00:00 2001
From: Pietro Albini <pietro@pietroalbini.org>
Date: Fri, 31 Jan 2020 11:02:34 +0100
Subject: [PATCH 2/2] Runner.Listener: prevent self-updates

There is surely a better way to do this, but hey, this hack might be
enough.
---
 src/Runner.Listener/SelfUpdater.cs | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/Runner.Listener/SelfUpdater.cs b/src/Runner.Listener/SelfUpdater.cs
index 92ed0ce..89bfec3 100644
--- a/src/Runner.Listener/SelfUpdater.cs
+++ b/src/Runner.Listener/SelfUpdater.cs
@@ -61,6 +61,13 @@ namespace GitHub.Runner.Listener
 
                 Trace.Info($"An update is available.");
 
+                // disable self-updates
+                var rustAvoidUnreachableCodeError = true;
+                if (rustAvoidUnreachableCodeError) {
+                    Trace.Info("prevented self-update");
+                    return false;
+                }
+
                 // Print console line that warn user not shutdown runner.
                 await UpdateRunnerUpdateStateAsync("Runner update in progress, do not shutdown runner.");
                 await UpdateRunnerUpdateStateAsync($"Downloading {_targetPackage.Version} runner");
-- 
2.31.1

